crates/nu-command/src/conversions/into/duration.rs

```diff
@@ -236,22 +242,25 @@ fn action(input: &Value, unit: &str, span: Span) -> Value {
     let value_span = input.span();
     match input {
         Value::Duration { .. } => input.clone(),
-        Value::String { val, .. } => match compound_to_duration(val, value_span) {
-            Ok(val) => Value::duration(val, span),
-            Err(error) => Value::error(error, span),
-        },
+        Value::String { val, .. } => {
+            if let Ok(num) = val.parse::<f64>() {
+                let ns = unit_to_ns_factor(unit);
+                return Value::duration((num * (ns as f64)) as i64, span);
+            } else if let Ok(num) = val.parse::<i64>() {
+                let ns = unit_to_ns_factor(unit);
+                return Value::duration(num * ns, span);
```

Comment:
I think `parse::<f64>` will succeed for all cases for which `parse::<i64>` will succeed?

Reply:
yes, right
Removed

Reply:
In theory the int parse could permit more significant digits but that may be marginal for now.